sklearn/utils/estimator_checks.py

```diff
@@ -197,6 +198,9 @@ def _yield_transformer_checks(name, transformer):
         yield check_transformer_data_not_an_array
     # these don't actually fit the data, so don't raise errors
     yield check_transformer_general
+    # it's not important to preserve types with Clustering
```

Comment:
This need to be more specific why this is not important :)

Reply:
It's more that we can't preserve it, when transform return the cluster ID.